lsp/nls/src/requests/completion.rs

```diff
@@ -379,6 +424,43 @@ fn collect_record_info(
         .unwrap_or_default()
 }
 
+/// As we traverse a nested record all the way to the topmost record,
+/// accumulate all the record metadata and corresponding path.
+fn accumulate_record_meta_data<'a>(
+    record: ItemId,
+    info @ ComplInfo {
+        linearization,
+        lin_cache,
+    }: &'a ComplInfo<'a>,
+    mut path: Vec<Ident>,
+    mut result: Vec<(&'a LinearizationItem<Types>, Vec<Ident>)>,
+) -> Vec<(&'a LinearizationItem<Types>, Vec<Ident>)> {
```

Comment:
Is it useful to have a return value? Aren't we mutating the argument directly?

Reply:
We're mutating it locally. I'm mostly using it as accumulator here. But we can also pass in it as a mutable reference, and not return anything.

Reply:
I mean, it seems like you never actually use the return value, so I think it makes more sense to return `unit`, yeah.